#### First Nural network to classify a Diabatic Class based on a simple Nural Network on PIMA Diabetic Datset

In [1]:
import pandas as pd

In [8]:
from keras.models import Sequential
from keras.layers import Dense

#### Read csv and create a Data Frame

In [9]:
pima_df = pd.read_csv('./Downloads/pima-indians-diabetes_data.csv',names=['Number of times pregnant','Plasma glucose concentration','Diastolic blood pressure (mm Hg)','Triceps skin fold thickness (mm)','2-Hour serum insulin (mu U/ml)',' Body mass index','Diabetes pedigree function','Age','Class'])

In [10]:
pima_df

Number of times pregnant  Plasma glucose concentration  ...  Age  Class
0                           6                           148  ...   50      1
1                           1                            85  ...   31      0
2                           8                           183  ...   32      1
3                           1                            89  ...   21      0
4                           0                           137  ...   33      1
..                        ...                           ...  ...  ...    ...
763                        10                           101  ...   63      0
764                         2                           122  ...   27      0
765                         5                           121  ...   30      0
766                         1                           126  ...   47      1
767                         1                            93  ...   23      0

[768 rows x 9 columns]

#### Create Independent and Dependent variables Set

In [15]:
Y=pima_df[['Class']] # Dependent variable
X=pima_df.iloc[:,0:8] # Independent variable

#### Create Test and Training data

In [18]:
from sklearn.model_selection import train_test_split

x_train,x_test,y_train,y_test = train_test_split(X,Y,test_size = 0.3)

#### Create Nural network Model 

In [19]:
model=Sequential()
model.add(Dense(12,input_dim=8,activation='relu'))
model.add(Dense(8,activation='relu'))
model.add(Dense(1,activation='sigmoid'))

#### Compile the Model and fit the test data

In [20]:
model.compile(loss='binary_crossentropy',optimizer='adam',metrics=['accuracy'])

In [22]:
model.fit(x_train,y_train,epochs=150,batch_size=10,verbose=0)

#### predict dependent variable based on test set

In [23]:
y_pred = model.predict_classes(x_test)

Instructions for updating:
Please use instead:* `np.argmax(model.predict(x), axis=-1)`,   if your model does multi-class classification   (e.g. if it uses a `softmax` last-layer activation).* `(model.predict(x) > 0.5).astype("int32")`,   if your model does binary classification   (e.g. if it uses a `sigmoid` last-layer activation).


#### Print Test set and Predicted set values

In [24]:
y_test.head(5),y_pred[0:5]

(     Class
 303      1
 588      1
 410      0
 160      0
 441      0,
 array([[0],
        [1],
        [0],
        [0],
        [0]]))

#### Check Accuracy Score of the model and print confusion matrix

In [28]:
from sklearn.metrics import accuracy_score, confusion_matrix

In [26]:
acc_score = accuracy_score(y_test,y_pred)
acc_score

0.70995670995671

In [29]:
confusion_matrix(y_test,y_pred)

array([[136,  19],
       [ 48,  28]], dtype=int64)

##### Analyzing the confusion matrix
True Positives (TP): we correctly predicted that they do have diabetes 28

True Negatives (TN): we correctly predicted that they don't have diabetes 136

False Positives (FP): we incorrectly predicted that they do have diabetes (a "Type I error") 19 Falsely predict positive Type I error

False Negatives (FN): we incorrectly predicted that they don't have diabetes (a "Type II error") 48 Falsely predict negative Type II error

In [40]:
y_test.groupby(['Class']).size()  ## Count to validate confusion Matrix

Class
0    155
1     76
dtype: int64